src/language-server/project/rover/DocumentSynchronization.ts

```diff
@@ -37,7 +44,7 @@ export function handleFilePartUpdates(
         newOffset === oldPart.source.locationOffset.line + offsetCorrection)
     ) {
       // replacement of chunk
-      newParts.push({ source, fractionalIndex: oldPart.fractionalIndex });
+      newParts.push({ ...oldPart, source });
```

Comment:
Keeping the old file part around and just updating the new `source`. This is important as file parts now have an additional `diagnostics` property that should persist user changes until the upstream LSP gets to reporting new diagnostics.